### PR TITLE
Use custom plugin to publish zips to maven

### LINKED ISSUE #378: Publish plugin zips to maven repo
### Description

With existing setup the plugin jars are only published to maven repo, but not the generated plugin zips.

### Related META issue

https://github.com/opensearch-project/opensearch-build/issues/1916

### Current Behavior

Existing all maven publications target the jar files to publish to maven, but not the generated distribution zips.

### Expected Behavior

Along with plugin jars, plugin zips should also be published to maven repo with right maven coordinates, so user can download these plugins using dependancy model.

### Steps To Reproduce

1. Run the gradle assemble task that outputs jar and zips.
2. Using build setup the jars are published to local staging maven repo.
3. Zip's are just added to distribution's folder during build runtime.
4. Finally once build is done, just the jar files are added to local maven repo and updates to the manifest file, but not the zips.

### Proposed solution
A custom gradle plugin solution that can create the publication task for the zips to get published to local staging maven repo with right maven coordinates.
Add/Configure the custom gradle plugin openseearch.pluginzip to the gradle project and test the generated plugin zip's to upload to local maven repo.
More details on how to use the Plugin:
https://github.com/opensearch-project/opensearch-plugins/blob/main/BUILDING.md#opensearchpluginzip

### PR DESCRIPTION
**Pull request by martin-gaievski**
Signed-off-by: Martin Gaievski <gaievski@amazon.com>

### Description
Follow [campaign](https://github.com/opensearch-project/opensearch-build/issues/1916) and use custom plugin to publish zips to maven
 
### Issues Resolved
#378 
 
### Check List
- [X] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: settings.gradle [hunk 1]
```diff
@@ -8,3 +8,5 @@ rootProject.name = 'opensearch-knn'
 include ":qa"
 include ":qa:rolling-upgrade"
 include ":qa:restart-upgrade"
+
+startParameter.excludedTaskNames=["publishPluginZipPublicationToMavenLocal"]
```
Comment (prudhvigodithi) [inline settings.gradle:12]: Hey we dont need to exclude this task as in build.sh we dont call `publishAllPublicationsToStagingRepository` 
https://github.com/opensearch-project/opensearch-plugins/blob/main/BUILDING.md#opensearchpluginzip (Point 7)

File: scripts/build.sh [hunk 2]
```diff
@@ -100,6 +100,7 @@ make opensearchknn_faiss opensearchknn_nmslib
 
 cd $work_dir
 ./gradlew assemble --no-daemon --refresh-dependencies -DskipTests=true -Dopensearch.version=$VERSION -Dbuild.snapshot=$SNAPSHOT -Dbuild.version_qualifier=$QUALIFIER
+./gradlew publishPluginZipPublicationToZipStagingRepository -Dopensearch.version=$VERSION -Dbuild.snapshot=$SNAPSHOT -Dbuild.version_qualifier=$QUALIFIER
```
Comment (prudhvigodithi) [inline scripts/build.sh:104]: Please add `cp` command also example as
https://github.com/opensearch-project/job-scheduler/blob/main/scripts/build.sh#L78-L79

File: build.gradle [hunk 3]
```diff
@@ -80,6 +81,29 @@ allprojects {
...
+                licenses {
+                    license {
+                        name = "The Apache License, Version 2.0"
+                        url = "http://www.apache.org/licenses/LICENSE-2.0.txt"
+                    }
+                }
+                developers {
+                    developer {
+                        name = "OpenSearch"
+                        url = "https://github.com/opensearch-project/opensearch-plugin-template-java"
```
Comment (prudhvigodithi) [inline build.gradle:99]: url should be url = "https://github.com/opensearch-project/k-NN"

File: settings.gradle (hunk 1 above)
Comment (martin-gaievski) [reply settings.gradle:12]: ack

File: build.gradle (hunk 3 above)
Comment (martin-gaievski) [reply build.gradle:99]: ack

File: scripts/build.sh (hunk 2 above)
Comment (martin-gaievski) [reply scripts/build.sh:104]: ack, added

**Comment by prudhvigodithi** [comment]
Hey @martin-gaievski
LGTM, please test to run `build.sh` on local to verify this works and able to publish the zip to local maven repo.
Thank you

**Review by naveentatikonda — APPROVED** [review]

**Pull request merged**